docs/speech-usage.rst

```diff
@@ -145,5 +141,81 @@ words to the vocabulary of the recognizer.
     transcript: Hello, this is a test
     confidence: 0.81
 
+
+Streaming Recognition
+---------------------
+
+The :meth:`~google.cloud.speech.Client.stream_recognize` method converts speech
+data to possible text alternatives on the fly.
+
+.. note::
+    Streaming recognition requests are limited to 1 minute of audio.
+
+    See: https://cloud.google.com/speech/limits#content
+
+.. code-block:: python
+
+    >>> from google.cloud import speech
+    >>> client = speech.Client()
+    >>> with open('./hello.wav', 'rb') as stream:
+    ...     sample = client.sample(stream=stream, encoding=speech.Encoding.LINEAR16,
+    ...                            sample_rate=16000)
+    ...     for response in client.stream_recognize(sample):
+    ...         print(response.transcript)
+    ...         print(response.is_final)
+    hello
+    True
```

Comment:
Continuing from [before][1] it might be worth having a `stream_recognize_multi` and `stream_recognize` which returns a single response?

Or you could just consume the whole stream and leave a note about why only one:

```python
>>> with open('./hello.wav', 'rb') as stream:
...     sample = client.sample(stream=stream,
...                            encoding=speech.Encoding.LINEAR16,
...                            sample_rate=16000)
...     responses = list(client.stream_recognize(sample))
...     # Only expect one result since `interim_results=False` by default
...     print(len(responses))
...     print(responses[0].transcript)
...     print(responses[0].is_final)
``` 

[1]: https://github.com/GoogleCloudPlatform/google-cloud-python/pull/2523#discussion_r83891167